[WebGL] Make WebGL via Metal internal

djg opened the pull request:
#### 4929b4c592f9bebd8079fc1ad5c9a691d60f827f
<pre>
[WebGL] Make WebGL via Metal internal
<a href="https://bugs.webkit.org/show_bug.cgi?id=250433">https://bugs.webkit.org/show_bug.cgi?id=250433</a>
rdar://problem/104104772

Reviewed by Brent Fulgham.

Due to sandboxing, in the post-Ventura world it&apos;s becoming harder and harder to
run WebGL via OpenGL instead of Metal and is not generally useful preference for
developers to toggle.

Change the status of `WebGLUsingMetal` to `internal` to reflect this.

Canonical link: <a href="https://commits.webkit.org/265257@main">https://commits.webkit.org/265257@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a619079afad6c048f18406655468f2262ce3cedc

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/10291 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/10528 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/10794 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/11937 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/9903 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/12522 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/10478 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/12851 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/10448 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/11178 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/8672 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/12324 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/8486 "Passed tests") | | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/16598 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/8698 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/9584 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/9472 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/12720 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/9767 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/9921 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/8051 "3 failures") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/10424 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/9080 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/2698 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/2499 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/13330 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/10708 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/9761 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/2643 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

brentfulgham reviewed (approved):
LGTM.

brentfulgham commented on Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
```diff
@@ -6543,7 +6543,7 @@ WebGLEnabled:
 WebGLUsingMetal:
   type: bool
-  status: testable
+  status: internal
```
Note that this change means that this toggle will not be flipped on automatically for test runs; it seems like this isn't an issue, since the feature will be enabled by default on platforms that support WebGL via Metal.

djg committed:
[WebGL] Make WebGL via Metal internal
https://bugs.webkit.org/show_bug.cgi?id=250433
rdar://problem/104104772

Reviewed by Brent Fulgham.

Due to sandboxing, in the post-Ventura world it's becoming harder and harder to
run WebGL via OpenGL instead of Metal and is not generally useful preference for
developers to toggle.

Change the status of `WebGLUsingMetal` to `internal` to reflect this.

Canonical link: https://commits.webkit.org/265257@main
